qa/rpc-tests/test_framework/util.py

```diff
@@ -355,33 +355,46 @@ def initialize_chain(test_dir,bitcoinConfDict=None,wallets=None, bins=None):
     4 wallets.
     """
 
+    logOn = os.getenv("PYTHON_DEBUG")
+    if logOn=="ERROR":
+        logging.getLogger().setLevel(logging.ERROR)
+    if logOn=="WARN":
+        logging.getLogger().setLevel(logging.WARN)
+    if logOn=="DEBUG":
+        logging.getLogger().setLevel(logging.DEBUG)
+    if logOn == "1" or logOn == "INFO":
+        logging.getLogger().setLevel(logging.INFO)
+
     if (not os.path.isdir(os.path.join("cache","node0"))
         or not os.path.isdir(os.path.join("cache","node1"))
         or not os.path.isdir(os.path.join("cache","node2"))
         or not os.path.isdir(os.path.join("cache","node3"))):
 
+        logging.info("Creating 200 block testnet block blockchain in 'cache' directory")
         #find and delete old cache directories if any exist
         for i in range(4):
             if os.path.isdir(os.path.join("cache","node"+str(i))):
                 shutil.rmtree(os.path.join("cache","node"+str(i)))
+        for i in range(4):
+            if os.path.isdir(os.path.join("building_cache","node"+str(i))):
+                shutil.rmtree(os.path.join("building_cache","node"+str(i)))
 
         # Create cache directories, run bitcoinds:
+        logging.info("  starting 4 bitcoind")
         for i in range(4):
             for retry in range(4):
                 try:
-                    datadir=initialize_datadir("cache", i, bitcoinConfDict, wallets, bins)
+                    datadir=initialize_datadir("building_cache", i, bitcoinConfDict, wallets, bins)
```

Comment:
I don't get the change to "building_cache" ? was "cache" giving a problem?

Reply:
Yes, if cache construction was aborted then all subsequent runs would use the bad cache.  The result is weird failures of tests that are intolerant of a < 200 block cache.